src/main/java/com/drajer/bsa/auth/impl/RestApiAuthorizerImpl.java

```diff
@@ -0,0 +1,29 @@
+package com.drajer.bsa.auth.impl;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.env.Environment;
+import org.springframework.http.HttpHeaders;
+
+import com.drajer.bsa.auth.RestApiAuthorizationHeaderIf;
+import com.drajer.bsa.model.KarProcessingData;
+
+public class RestApiAuthorizerImpl implements RestApiAuthorizationHeaderIf {
+    private final Logger logger = LoggerFactory.getLogger(SampleRestApiAuthorizer.class);
+
+    @Autowired private Environment environment;
+
+    @Override
+    public HttpHeaders getAuthorizationHeader(KarProcessingData data) {
+
+        HttpHeaders headers = new HttpHeaders();
+        logger.info("Fetching the AccessToken");
+
+        String securityToken = environment.getRequiredProperty("ecr.auth.token");
+
+        // Set Authorization Header
+        headers.add("Authorization", "Bearer " + securityToken);
```

Comment:
We could use the `X-Api-Key` header to pass the security token since Ehrv8ExtnService already has a filter to authenticate requests by that header value.
[ehrv8-extension-dotnet-core-api/EHRV8ExtnAPIServices/Filters/ApiKeyAuthenticationHandler.cs](https://github.com/RXNT/ehr-apps/blob/development/ehrv8-extension-dotnet-core-api/EHRV8ExtnAPIServices/Filters/ApiKeyAuthenticationHandler.cs)

Here is an example of how we could use custom authentication schemes in controllers. 
https://github.com/RXNT/ehr-apps/blob/4bab4c39795e81aae70ebb2ce1c0ccc60e3bed8a/ehrv8-extension-dotnet-core-api/EHRV8ExtnAPIServices/Controllers/ElectronicCaseReportingController.cs#L19